src/type_checker.cpp

```diff
@@ -198,7 +197,8 @@ void TypeChecker::Visit(IdLabeledStmtNode& id_labeled_stmt) {
   // TODO: To allow forward reference, the label has to be added in some other
   // way.
   // TODO: The label has function scope.
-  env_.Add(std::make_unique<SymbolEntry>(id_labeled_stmt.label));
+  auto label_type = Type();
+  env_.Add(std::make_unique<SymbolEntry>(id_labeled_stmt.label, label_type));
```

Comment:
Inline the variable since its already clear that its the type of the label.
Addionally, having labels as error type is unusual. I suggest adding a comment.
```suggestion
  // XXX: Labels have error type.
  env_.Add(std::make_unique<SymbolEntry>(id_labeled_stmt.label, PrimitiveType::kError));
```